### ISSUE TITLE
Missing get-*-point (i.e. get_val) methods with MPI

### ISSUE THREAD

**Issue by thchr** [NanoComp/mpb#100]
The otherwise very useful `(get-*-point r)` methods, e.g. `(get-field-point r)`, do not work with MPI since the method `get_val` doesn't have an MPI implementation. 

This is already nicely noted by an error message:

https://github.com/NanoComp/mpb/blob/47f68c2df1f66d80bd17433d80c6814d15158070/mpb/fields.c#L571-L572

How painful would it be to implement this functionality? I'm assuming the difficulty lies in locating which local process owns the `ix` index? 

This is pretty much guesswork on my part, as I don't know MPI, but, conceptually, would this involve something like:
```c
int ixp = ix - local_x_start
if (ixp >= 0 && ixp < local_nx)
     // assign data[(((ixp * ny) + iy) * nz + iz) * stride] to a process buffer
else
     // assign zero to the (same) process buffer
// reduce over buffer and return value
```
?

Best regards,
Thomas

**Comment by stevengj** [comment]
Yes, it would basically look like that.  Conceptually, it's not difficult, but it's a little annoying to get the coordinates right and debug, so I never got around to it, especially since I don't use the MPI version myself very much.

**Comment by stevengj** [comment]
Note, however, that you need to use `local_y_start` etcetera because the position-space fieldsare [transposed for performance reasons](http://www.fftw.org/fftw3_doc/Transposed-distributions.html#Transposed-distributions)  (x and y are swapped) .

**Comment by stevengj** [comment]
Closed by #112.
